### ISSUE TITLE
First Render Always Blank

### ISSUE THREAD

**Issue by skudirka-tw** [GetStream/stream-chat-react-native#2037]
**Describe the bug**
The `OverlayProvider` blocks rendering content on the very first render because it starts registering the language in a useEffect in the `useStreami18n` hook. Initially we had the OverlayProvider wrap a react navigation stack (since we had multiple screens with your UI components). But because the user would see a white flash when navigating to this stack, we decided to wrap each individual screen content instead. You still see a white flash in the content area, but at least the screen headers render right away. 

I think this could be done sooner and not have to wait for the component to mount. Or at least provide a loader to render in its place if this cannot be avoided.

| OverlayProvider | useStreami18n |
| ------ | ------ |
|    ![image](https://user-images.githubusercontent.com/115643473/229579481-d9f4f2f5-94bb-4684-9480-2e040aba0f4d.png)    |    ![image](https://user-images.githubusercontent.com/115643473/229580547-0b196102-c9f7-463f-b304-0dda07c5cc74.png)  |

**Dev environment info (please complete/provide the following information):**

- Package version stream-chat-react-native@5.12.1-beta.1
- react-native
- Device/Emulator: Android and IOS

**To Reproduce**

1. Render any content within the OverlayProvider and see you get a flash of blank content on first render.

**Expected behavior**
Content should render right away, or have placeholder content (like a loader) instead.

**Screenshots**
Screenshots above. 

Here is a video as well when we had the navigation header wrapped in the OverlayProvider. See the delayed render going to the Messages screen:

https://user-images.githubusercontent.com/115643473/229585359-60fcae28-6b2d-48ca-8e4b-96d7fde549dc.mp4

**Comment by vanGalilea** [comment]
It's good to know we have a PR that will solve this issue, see history line above.

**Comment by vanGalilea** [comment]
Hey @skudirka-tw I have tried to reproduce this issue, yet with no success.

Have a look at my triaging journey while:
- adding `seTimeout` of 3000 s when setting the language.
- the pink view represents anything that will be in the `OverlayProvider`.
- the green view represents anything that will be outside `OverlayProvider`

https://user-images.githubusercontent.com/25864161/234872877-41ac6983-9ac3-45dc-8aff-fa8cab8d60d3.mp4

Although we haven't managed to reproduce this issue, we have refactored the related i18n code with the suggestions you've made.
